### PR TITLE
Adds support for Jobs

### LINKED ISSUE #30: Support validation of Jobs
Enable admission controller validation for Job objects.

For more information on Jobs, see https://kubernetes.io/docs/concepts/workloads/controllers/job/.

### PR DESCRIPTION
**Pull request by zhill**
Adds Job support. Resolves #30

### CODE REVIEW

**Comment by luhring** [comment]
Closes #30

File: cmd/kubernetes-admission-controller/admission/hook_test.go [hunk 1]
```diff
@@ -101,6 +102,7 @@ func TestHook_Validate(t *testing.T) {
 			admissionRequest:      podAdmissionRequest(t, nonexistentImageName),
 			isExpectedToBeAllowed: true,
 		},
+		// Deployment resource tess
```
Comment (luhring) [inline cmd/kubernetes-admission-controller/admission/hook_test.go:105]: nit: sp. `tess` -> `tests`

File: cmd/kubernetes-admission-controller/extractor/from_job_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+var (
+	testJobObjectMeta = metav1.ObjectMeta{
+		Labels: map[string]string{
+			"key": "value",
+		},
+		Annotations: map[string]string{
+			"annotation1": "value1",
+		},
+		Name:      "a_deployment",
```
Comment (luhring) [inline cmd/kubernetes-admission-controller/extractor/from_job_test.go:45]: This shouldn't affect the tests, but my original idea was to have the names match the kind of object being defined, e.g. `a_job`. But don't worry about this, I have some other ideas to mitigate the copy/paste required for new tests.

**Review by luhring — APPROVED** [review]
Very nice work! 🎉  And it's very informative to see the pattern playing out with new kinds of resources. I've noticed a few opportunities for an improvement to what I had put in place in the last PR, to help reduce copy/paste needed.

**Comment by zhill** [comment]
I addressed feedback but updated the branch name to issue-30, so I'll close this and open a new PR from that branch where I also added the issue reference in the commit.

**Pull request closed without merging**